### PR TITLE
Add authenticated endpoint with tests

### LINKED ISSUE #15: Implement JWT Authentication
Implement JWT-based authentication and add it to all existing write (create, update, and delete) endpoints.

### PR DESCRIPTION
**Pull request by egiurleo**
This PR builds off @DanielKim1's work in #47 

Closes #15

The only change is that this PR adds two tests demonstrating that auth works, along with some new environment variables and info in the README.

### CODE REVIEW

File: tests/api/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def test_private_endpoint_no_auth(self):
+    response = self.client.get("/api/private")
+    self.assertEqual(response.status_code, 401)
+    json_response = json.loads(response.data.decode("utf-8"))
+
+    self.assertEqual(json_response["description"], "Authorization header is expected")
+  
+  @unittest.skipIf(not os.environ.get("AUTH0_CLIENT_ID") or not os.environ.get("AUTH0_CLIENT_SECRET"),
+                     "Cannot run test without AUTH0_CLIENT_ID and AUTH0_CLIENT_SECRET environment variables")
```
Comment (egiurleo) [inline tests/api/test_auth.py:21]: Skip the test if the AUTH0_CLIENT_ID and AUTH0_CLIENT_SECRET env variables are `None` or empty strings.

File: tests/api/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  def test_private_endpoint_auth(self):
+    data = {
+      "client_id": os.environ.get('AUTH0_CLIENT_ID'),
+      "client_secret": os.environ.get('AUTH0_CLIENT_SECRET'),
+      "audience": API_AUDIENCE,
+      "grant_type":"client_credentials"
+    }
+
+    jwt_response = requests.post(f'https://{AUTH0_DOMAIN}/oauth/token', data=data)
+    jwt = json.loads(jwt_response.text)['access_token']
```
Comment (egiurleo) [inline tests/api/test_auth.py:31]: I used this tutorial to figure out how to generate a JWT: https://auth0.com/blog/using-python-flask-and-angular-to-build-modern-web-apps-part-2/

**Review by JZenk — CHANGES_REQUESTED** [review]

File: tests/api/test_auth.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (egiurleo) [reply tests/api/test_auth.py:21]: @JZenk Good call! For some reason `requests` didn't end up in the `requirements.txt` file. Could you try re-running `pip install -r requirements.txt` and running the tests again?

[the remark this answers is not in the document]
Comment (egiurleo) [reply tests/api/test_auth.py:21]: And to be clear, please pull the changes I made first :)

[the remark this answers is not in the document]
Comment (JZenk) [reply tests/api/test_auth.py:21]: Huzzah! It now skips!

```bash
root@cc19da0e08ac:/app# python3 -m unittest
s.........................
----------------------------------------------------------------------
Ran 26 tests in 0.963s

OK (skipped=1)
```

**Review by JZenk — APPROVED** [review]
🏆 

I need to be added to Auth0 to fully test, but I am approving because the implementation is the same as the Auth0 howto.

Aka public APIs still work, private API fails as expected.

**Pull request merged**